src/runtime_src/xdp/profile/database/static_info_database.cpp

```diff
@@ -151,6 +151,63 @@ namespace xdp {
     }
   }
 
+  bool VPStaticDatabase::validXclbin(void* devHandle)
+  {
+    std::shared_ptr<xrt_core::device> device =
+      xrt_core::get_userpf_device(devHandle);
+
+    // If this xclbin was built with tools before the 2019.2 release, we
+    //  do not support profiling
+    std::pair<const char*, size_t> buildMetadata =
+      device->get_axlf_section(BUILD_METADATA);
+    const char* buildMetadataSection = buildMetadata.first;
+    size_t      buildMetadataSz      = buildMetadata.second;
+    if (buildMetadataSection != nullptr) {
+      std::stringstream ss ;
+      ss.write(buildMetadataSection, buildMetadataSz) ;
+
+      boost::property_tree::ptree pt;
+      boost::property_tree::read_json(ss, pt);
+
+      std::string version =
+	pt.get<std::string>("build_metadata.xclbin.generated_by.version", "") ;
+
+      if (version == "" || version.find(".") == std::string::npos)
+	return false ;
+
+      auto firstDotPosition = version.find(".") ;
```

Comment:
Instead of parsing the version string and then converting major and minor value separately into "int", we can use "strtod" function to convert the version string to a "double" value and compare it with 2019.2 double value.

Reply:
After verifying that std::stod works on strings with single decimal points and multiple periods, I have converted this code to a call to std::stod